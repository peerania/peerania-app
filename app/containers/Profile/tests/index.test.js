import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import configureStore from 'configureStore';
import createdHistory from 'createdHistory';

import EosioProvider from 'containers/EosioProvider';
import { Profile } from '../index';

const children = <div>Children</div>;
React.Children.only = jest.fn().mockImplementation(() => children);

// const cmp = new Profile();
const props = {
  children: null,
  userId: 'userId',
  profile: {
    profile: {},
  },
  locale: 'en',
  isProfileLoading: false,
  getUserProfileDispatch: jest.fn(),
};

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Profile', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, createdHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <EosioProvider>
          <Profile {...props} />
        </EosioProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  /*it('componentDidMount', () => {
    const userId = 'user111';

    cmp.props.userId = userId;
    cmp.componentDidMount();
    expect(cmp.props.getUserProfileDispatch).toHaveBeenCalledWith(userId, true);
  });

  describe('componentWillReceiveProps', () => {
    it('nextProps.userId !== this.props.userId', () => {
      const nextProps = {
        userId: 'user2222',
      };

      cmp.props.userId = 'user1111';

      cmp.componentWillReceiveProps(nextProps);
      expect(cmp.props.getUserProfileDispatch).toHaveBeenCalledWith(
        nextProps.userId,
        true,
      );
    });
  });*/
});
