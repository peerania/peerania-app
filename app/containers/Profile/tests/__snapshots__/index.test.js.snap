// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render and match the snapshot 1`] = `
<div>
  Children
</div>
`;
